test/jestsetup.js

```diff
@@ -5,3 +5,56 @@ configure({ adapter: new Adapter() })
 
 global.mount = mount
 global.shallow = shallow
+
+const isDeprecatedLifecycleWarning = (msg, componentName) => {
+  return (
+    msg.includes('has been renamed, and is not recommended for use') &&
+    msg.includes(componentName)
+  )
+}
+
+const ignoreOnConditions = (originalWarn, ignoreConditions) => {
+  return function(...args) {
+    const msg = args[0]
+    if (ignoreConditions.some(condition => condition(msg))) {
+      return
+    }
+    throw new Error(
+      'Code in tests should not emit any warning. Warning emitted:' + msg
+    )
```

Comment:
Hardcore… but ok I think, it's not too difficult to add exceptions and promotes good practice.

Reply:
In fact it was not meant to be in this commit. I prefer to remove it for now and add it later.